src/main/java/org/osc/manager/ism/api/IsmDeviceApi.java

```diff
@@ -138,80 +189,184 @@ public Void call() throws Exception {
     @Override
     public String createDeviceMember(final String name, String ipAddress, String vserverIpAddress, String contactIpAddress,
             String gateway, String prefixLength) throws Exception {
-        return txControl.required(new Callable<DeviceMember>() {
+
+        DeviceEntity device = new DeviceEntity();
+        DeviceMemberEntity deviceMember = new DeviceMemberEntity();
+        deviceMember.setParent(device);
+        device.setId(null);
+        deviceMember.setName(name);
+        return createDeviceMember(deviceMember);
+    }
+
+    public String createDeviceMember(DeviceMemberEntity entity) throws Exception {
+
+        return this.txControl.required(new Callable<DeviceMemberEntity>() {
+
             @Override
-            public DeviceMember call() throws Exception {
-                VSSDevice device = em.find(VSSDevice.class, vs.getId());
-                
-                DeviceMember dm = new DeviceMember();
-                dm.setParent(device);
-                dm.setName(name);
-                em.persist(dm);
-                return dm;
+            public DeviceMemberEntity call() throws Exception {
+
+                CriteriaBuilder criteriaBuilder = IsmDeviceApi.this.em.getCriteriaBuilder();
+                CriteriaQuery<DeviceEntity> query = criteriaBuilder.createQuery(DeviceEntity.class);
+                Root<DeviceEntity> r = query.from(DeviceEntity.class);
+                if (entity.getParent().getId() != null) {
+                    query.select(r)
+                            .where(criteriaBuilder.and(criteriaBuilder.equal(r.get("id"), entity.getParent().getId())));
+                }
```

Comment:
} else { //no  line breaks. Auto indent when saving the file on eclipse should take care of this.